app/context/ResourceContext/index.tsx

```diff
@@ -13,42 +13,57 @@ const REPORTS_BLOCK = [];
 const ADMIN_ACCESS = ['admin'];
 const ADMIN_BLOCK = ALL_ROLES;
 
-type UseResourcesReturnType = {
-  germlineAccess: boolean;
-  reportsAccess: boolean;
-  adminAccess: boolean;
-  reportSettingAccess: boolean;
-};
-
-const useResources = (): UseResourcesReturnType => {
-  const { userDetails } = useContext(SecurityContext);
+const useResources = (): ResourceContextType => {
+  const { userDetails: { groups, ident: userIdent } } = useSecurity();
+  const { report } = useContext(ReportContext);
 
   const [germlineAccess, setGermlineAccess] = useState(false);
   const [reportsAccess, setReportsAccess] = useState(false);
+  const [reportEditAccess, setReportEditAccess] = useState(false);
   const [adminAccess, setAdminAccess] = useState(false);
   const [reportSettingAccess, setReportSettingAccess] = useState(false);
 
+  // Check user group first to see which resources they can access
   useEffect(() => {
-    if (userDetails?.groups) {
-      if (checkAccess(userDetails.groups, GERMLINE_ACCESS, GERMLINE_BLOCK)) {
+    if (groups) {
+      if (checkAccess(groups, GERMLINE_ACCESS, GERMLINE_BLOCK)) {
         setGermlineAccess(true);
       }
 
-      if (checkAccess(userDetails.groups, REPORTS_ACCESS, REPORTS_BLOCK)) {
+      if (checkAccess(groups, REPORTS_ACCESS, REPORTS_BLOCK)) {
         setReportsAccess(true);
       }
 
-      if (checkAccess(userDetails.groups, ADMIN_ACCESS, ADMIN_BLOCK)) {
+      if (checkAccess(groups, ADMIN_ACCESS, ADMIN_BLOCK)) {
         setAdminAccess(true);
       }
-      if (checkAccess(userDetails.groups, [...ADMIN_ACCESS, 'manager'], ADMIN_BLOCK)) {
+
+      if (checkAccess(groups, [...ADMIN_ACCESS, 'manager'], ADMIN_BLOCK)) {
         setReportSettingAccess(true);
+        setReportEditAccess(true);
+      }
+    }
+  }, [groups]);
+
+  /**
+   * Check report specific permissions if user isn't admin
+   */
+  useEffect(() => {
```

Comment:
@elewis2 If there are any other details you would like to add here -- currently they can edit if they are added to the report, did we want to more specifically check the permissions?

Reply:
No, I think this makes sense for now.